core/domain/stats_services_test.py

```diff
@@ -14,20 +14,233 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
+
 from core.domain import event_services
 from core.domain import exp_domain
 from core.domain import exp_services
-from core.domain import stats_jobs_continuous
 from core.domain import stats_domain
+from core.domain import stats_jobs_continuous
 from core.domain import stats_services
 from core.domain import user_services
 from core.platform import models
 from core.tests import test_utils
 import feconf
+import utils
 
 (stats_models,) = models.Registry.import_models([models.NAMES.statistics])
 
 
+class StatisticsServicesTest(test_utils.GenericTestBase):
+    """Test the helper functions and methods defined in the stats_services
+    module.
+    """
+    def setUp(self):
+        super(StatisticsServicesTest, self).setUp()
+        self.exp_id = 'exp_id1'
+        self.exp_version = 1
+        self.stats_model_id = (
+            stats_models.ExplorationStatsModel.create('exp_id1', 1, 0, 0, {}))
+
+    def test_calls_to_stats_methods(self):
+        """Test that calls are being made to the
+        handle_stats_creation_for_new_exp_version and
+        handle_stats_creation_for_new_exploration methods when an exploration is
+        created or updated.
+        """
+        # Create exploration object in datastore.
+        exp_id = 'exp_id'
+        test_exp_filepath = os.path.join(
+            feconf.TESTS_DATA_DIR, 'string_classifier_test.yaml')
+        yaml_content = utils.get_file_contents(test_exp_filepath)
+        assets_list = []
+        with self.swap(feconf, 'ENABLE_ML_CLASSIFIERS', True):
+            exp_services.save_new_exploration_from_yaml_and_assets(
+                feconf.SYSTEM_COMMITTER_ID, yaml_content, exp_id,
+                assets_list)
+        exploration = exp_services.get_exploration_by_id(exp_id)
+
+        # Now, there are two analytics model in datastore (one from setup
+        # function).
+        all_models = stats_models.ExplorationStatsModel.get_all()
+        self.assertEqual(all_models.count(), 2)
+        exploration_stats = stats_services.get_exploration_stats_by_id(
+            exploration.id, exploration.version)
+        self.assertEqual(
+            exploration_stats.state_stats_mapping.keys(), ['Home', 'End'])
+
+        # Update exploration by adding a state.
+        change_list = [{
+            'cmd': 'add_state',
+            'state_name': 'New state'
+        }]
+        exploration.add_states(['New state'])
```

Comment:
Doesn't update_exploration below already take care of this?

Reply:
Yep. Was keeping it around to maintain consistency in domain object and datastore instance. Not really needed anymore.